drivers/can/mcp2515.c

```diff
@@ -0,0 +1,621 @@
+/*
+ * Copyright (c) 2018 Karsten Koenig
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <device.h>
+#include <spi.h>
+#include <gpio.h>
+
+#define LOG_LEVEL CONFIG_CAN_LOG_LEVEL
+#include <logging/log.h>
+LOG_MODULE_REGISTER(mcp2515_can);
+
+#include "mcp2515.h"
+
+static int mcp2515_cmd_soft_reset(struct device *dev)
+{
+	u8_t cmd_buf[] = { MCP2515_OPCODE_RESET };
+
+	const struct spi_buf tx_buf = {
+		.buf = cmd_buf, .len = sizeof(cmd_buf),
+	};
+	const struct spi_buf_set tx = {
+		.buffers = &tx_buf, .count = 1U
+	};
+
+	return spi_write(DEV_DATA(dev)->spi, &DEV_DATA(dev)->spi_cfg, &tx);
+}
+
+static int mcp2515_cmd_bit_modify(struct device *dev, u8_t reg_addr, u8_t mask,
+									u8_t data)
+{
+	u8_t cmd_buf[] = { MCP2515_OPCODE_BIT_MODIFY, reg_addr, mask, data };
+
+	const struct spi_buf tx_buf = {
+		.buf = cmd_buf, .len = sizeof(cmd_buf),
+	};
+	const struct spi_buf_set tx = {
+		.buffers = &tx_buf, .count = 1U
+	};
+
+	return spi_write(DEV_DATA(dev)->spi, &DEV_DATA(dev)->spi_cfg, &tx);
+}
+
+static int mcp2515_cmd_write_reg(struct device *dev, u8_t reg_addr,
+									u8_t *buf_data, u8_t buf_len)
```

Comment:
What has happened here with indentation?

It's messed up in numerous places in this file.

Reply:
Looks like you looked at an outdated version, probably what @alexanderwachter spotted and what was fixed by uncrustify, the alignments for the function headers look correct to me with 8 spaces tab.

Reply:
Sorry I was wrong about this, you looked at the commit adding the driver and not the full pull request. The commit with the driver was still faulty, the indentions were fixed in the sample commit, which was obviously wrong. 
Anyways, fixed the 80 char issues and also the indention and it's all properly in the first commit now that adds the driver